routes/habitRoutes.js

```diff
@@ -27,22 +27,11 @@ router.get('/habits', passport.authenticate('jwt', { session: false }), (req, re
  * @access  Private
  */
 router.post('/create', passport.authenticate('jwt', { session: false }), (req, res) => {
-	req.body.user = req.user._id.toString();
 	const { errors, isValid } = validateHabitsInput(req.body);
 	// Validate request body
 	if (!isValid) return res.status(400).json(errors);
 
-	const { user, name, description, type, difficulty, tags, frequency } = req.body;
-
-	return new Habit({
-		user,
-		name,
-		type,
-		description,
-		tags,
-		difficulty,
-		frequency,
-	})
+	return new Habit(Object.assign(req.body, { user: req.user._id.toHexString() }))
```

Comment:
Cool use of `assign` 😄 This can be shortened to:
`return new Habit({...req.body, user: req.user._id.toHexString()})`

but this works fine too.

Reply:
Old school spread 😄 Yeah, I actually had that first but I got:
`Parsing error: Unexpected token ... (undefined)`
for some reason. 🤔 But it does work so I'll switch it back.